### PR TITLE
chore: improve billing activation/deactivate tests and view validation

### PR DESCRIPTION
**Pull request by zlwaterfield**
## Changes

This PR is improving the activation/deactivate views to be better prepared for the subscribe to all products changes. It's 
- adding better validation, 
- deprecating `plan` query param
- adding testing for billing

This is using more of a [Django Styleguide](https://github.com/HackSoftware/Django-Styleguide) pattern for views

siblling PR in billing: https://github.com/PostHog/billing/pull/669

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Yes - will any self hosted users be affected by the removal of plan? Should we add a custom error if they pass that in?

## How did you test this code?

Add unit tests

### CODE REVIEW

File: ee/api/billing.py [hunk 1]
```diff
@@ -87,46 +87,49 @@ def patch(self, request: Request, *args: Any, **kwargs: Any) -> Response:
...
+        redirect_path = serializer.validated_data.get("redirect_path", "organization/billing")
         if redirect_path.startswith("/"):
             redirect_path = redirect_path[1:]
 
         redirect_uri = f"{settings.SITE_URL or request.headers.get('Host')}/{redirect_path}"
         url = f"{BILLING_SERVICE_URL}/activation?redirect_uri={redirect_uri}&organization_name={organization.name}"
 
-        plan = request.GET.get("plan", None)
-        product_keys = request.GET.get("products", None)
-        if not plan and not product_keys:
```
Comment (zlwaterfield) [inline ee/api/billing.py:104]: `plan` is no longer supported - can we remove it entirely, or should we keep it with a custom error?

Comment (xrdt) [reply ee/api/billing.py:104]: I feel like I keep running into situations where I'm confident something is fully removed and get proven wrong. I'd say leave it in with an error so we can track in sentry where the stragglers are.

File: ee/api/test/test_billing.py [hunk 2]
```diff
@@ -899,3 +899,64 @@ def mock_implementation(url: str, headers: Any = None, params: Any = None) -> Ma
...
+
+        self.assertIn("/activation", response.url)
+        self.assertIn("products=product_1:plan_1,product_2:plan_2", response.url)
+        url_pattern = r"redirect_uri=http://[^/]+/custom/path"
+        self.assertRegex(response.url, url_pattern)
+
+    def test_activation_with_default_redirect_path(self):
+        url = "/api/billing-v2/activation"
+        data = {
+            "products": "product_1:plan_1,product_2:plan_2",
```
Comment (xrdt) [inline ee/api/test/test_billing.py:920]: Worth testing if the serializer catches a bad request w/o products specified?

**Review by xrdt — APPROVED** [review]

File: ee/api/billing.py [hunk 3]
```diff
@@ -87,46 +87,67 @@ def patch(self, request: Request, *args: Any, **kwargs: Any) -> Response:
...
+
+            return data
+
     @action(methods=["GET"], detail=False)
     def activation(self, request: Request, *args: Any, **kwargs: Any) -> HttpResponse:
         license = get_cached_instance_license()
         organization = self._get_org_required()
 
-        redirect_path = request.GET.get("redirect_path") or "organization/billing"
+        serializer = self.ActivationSerializer(data=request.GET)
```
Comment (raquelmsmith) [inline ee/api/billing.py:117]: Rename to `Activate` since you did that in Billing?

Comment (zlwaterfield) [reply ee/api/billing.py:117]: Yep, that'll be on another PR when I move the URL in posthog as well. Doing that after these others are merged.

**Pull request merged**